client-side/js/options/colorPallet.js

```diff
@@ -0,0 +1,100 @@
+        input:hover {
+          border-color: grey;
+        }
+        #white {background-color: white;}
+        #black {background-color: black;}
+        #red {background-color: rgb(255, 0, 0);}
+        #orange {background-color: rgb(255, 111, 0);}
+        #yellow {background-color: rgb(255, 247, 0);}
+        #green {background-color: rgb(53, 189, 22);}
+        #cyan {background-color: rgb(0, 255, 212);}
+        #blue {background-color: rgb(0, 145, 255);}
+        #purple {background-color: rgb(187, 0, 255);}
+        #pink {background-color: rgb(255, 89, 177);}
+      </style>
+
+      <div class="pallet">
+        <input type="button" id="white">
+        <input type="button" id="black">
+        <input type="button" id="red">
+        <input type="button" id="orange">
+        <input type="button" id="yellow">
+        <input type="button" id="green">
+        <input type="button" id="cyan">
+        <input type="button" id="blue">
+        <input type="button" id="purple">
+        <input type="button" id="pink">
+      </div>
+    `
+    const white = element.querySelector('#white')
+    white.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(255, 255, 255)'
+      e.ctx.fillStyle = 'rgb(255, 255, 255)'
+    })
+    const black = element.querySelector('#black')
+    black.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(0, 0, 0)'
+      e.ctx.fillStyle = 'rgb(0, 0, 0)'
+    })
+    const red = element.querySelector('#red')
+    red.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(255, 0, 0)'
+      e.ctx.fillStyle = 'rgb(255, 0, 0)'
+    })
+    const orange = element.querySelector('#orange')
+    orange.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(255, 111, 0)'
+      e.ctx.fillStyle = 'rgb(255, 111, 0)'
+    })
+    const yellow = element.querySelector('#yellow')
+    yellow.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(255, 247, 0)'
+      e.ctx.fillStyle = 'rgb(255, 247, 0)'
+    })
+    const green = element.querySelector('#green')
+    green.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(53, 189, 22)'
+      e.ctx.fillStyle = 'rgb(53, 189, 22)'
+    })
+    const cyan = element.querySelector('#cyan')
+    cyan.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(0, 255, 212)'
+      e.ctx.fillStyle = 'rgb(0, 255, 212)'
+    })
+    const blue = element.querySelector('#blue')
+    blue.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(0, 145, 255)'
+      e.ctx.fillStyle = 'rgb(0, 145, 255)'
+    })
+    const purple = element.querySelector('#purple')
+    purple.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(187, 0, 255)'
+      e.ctx.fillStyle = 'rgb(187, 0, 255)'
+    })
+    const pink = element.querySelector('#pink')
+    pink.addEventListener('click', () => {
+      e.ctx.strokeStyle = 'rgb(255, 89, 177)'
+      e.ctx.fillStyle = 'rgb(255, 89, 177)'
+    })
```

Comment:
Anytime u have repetitive code like this, there's usually a better way to write it which would reduce redundancy && make the code easier to update in the future, the trick usually has something to do with creating a loop. ask urself:
> "what are the only bits of code that change? how can i organize them in a data structure? && how could i loop through that data structure to achieve the same logic i've currently written out manually?"

in this case, we could get rid of the id's in our `<input type="button">` elements (to avoid the issue mentioned above) && store each of those color strings in an array. we can then loop through all the input elements (by selecting all the elements in our `element` parent object which match the CSS query string `input[type="button"]`, ie. our buttons), then we loop through each of those input buttons, && for each of those change their backgroundColor to match the next item in the colors array, as well as add a "click" event listener to each.

```js
    const colors = [
      'rgb(255, 255, 255)', // white
      'rgb(0, 0, 0)', // black
      'rgb(255, 0, 0)', // red
      'rgb(255, 111, 0)', // orange
      'rgb(255, 247, 0)', // yellow
      'rgb(53, 189, 22)', // green
      'rgb(0, 255, 212)', // cyan
      'rgb(0, 145, 255)', // blue
      'rgb(187, 0, 255)', // purple
      'rgb(255, 89, 177)' // pink
    ]

    // select every <input type="button"> element...
    element.querySelectorAll('input[type="button"]').forEach((btn, i) => {
      // ...&& then "for each" of them (this is essentially a "for" loop)...
      // ...change their backgroundColor to match the next color on the list...
      btn.style.backgroundColor = colors[i]
      // ...then add a "click" event listener to update the canvas to match
      btn.addEventListener('click', () => {
        e.ctx.strokeStyle = colors[i]
        e.ctx.fillStyle = colors[i]
      })
    })
```

NOTE: the code above also assumes u've removed the `id` of each of those input elements as well as got rid of the id selector CSS rules (that were changing each's background-color) in ur styles.